// @genkit-ai/googleai/lib/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var src_exports = {};
__export(src_exports, {
  default: () => src_default,
  gemini: () => import_gemini.gemini,
  gemini10Pro: () => import_gemini.gemini10Pro,
  gemini15Flash: () => import_gemini.gemini15Flash,
  gemini15Flash8b: () => import_gemini.gemini15Flash8b,
  gemini15Pro: () => import_gemini.gemini15Pro,
  gemini20Flash: () => import_gemini.gemini20Flash,
  gemini20FlashExp: () => import_gemini.gemini20FlashExp,
  gemini20FlashLite: () => import_gemini.gemini20FlashLite,
  gemini20ProExp0205: () => import_gemini.gemini20ProExp0205,
  gemini25FlashPreview0417: () => import_gemini.gemini25FlashPreview0417,
  gemini25ProExp0325: () => import_gemini.gemini25ProExp0325,
  gemini25ProPreview0325: () => import_gemini.gemini25ProPreview0325,
  googleAI: () => googleAI,
  textEmbedding004: () => import_embedder.textEmbedding004,
  textEmbeddingGecko001: () => import_embedder.textEmbeddingGecko001
});
module.exports = __toCommonJS(src_exports);
var import_plugin = require("genkit/plugin");
var import_embedder = require("./embedder.js");
var import_gemini = require("./gemini.js");
function googleAI(options) {
  return (0, import_plugin.genkitPlugin)("googleai", async (ai) => {
    let apiVersions = ["v1"];
    if (options?.apiVersion) {
      if (Array.isArray(options?.apiVersion)) {
        apiVersions = options?.apiVersion;
      } else {
        apiVersions = [options?.apiVersion];
      }
    }
    if (apiVersions.includes("v1beta")) {
      Object.keys(import_gemini.SUPPORTED_V15_MODELS).forEach(
        (name) => (0, import_gemini.defineGoogleAIModel)({
          ai,
          name,
          apiKey: options?.apiKey,
          apiVersion: "v1beta",
          baseUrl: options?.baseUrl,
          debugTraces: options?.experimental_debugTraces
        })
      );
    }
    if (apiVersions.includes("v1")) {
      Object.keys(import_gemini.SUPPORTED_V1_MODELS).forEach(
        (name) => (0, import_gemini.defineGoogleAIModel)({
          ai,
          name,
          apiKey: options?.apiKey,
          apiVersion: void 0,
          baseUrl: options?.baseUrl,
          debugTraces: options?.experimental_debugTraces
        })
      );
      Object.keys(import_gemini.SUPPORTED_V15_MODELS).forEach(
        (name) => (0, import_gemini.defineGoogleAIModel)({
          ai,
          name,
          apiKey: options?.apiKey,
          apiVersion: void 0,
          baseUrl: options?.baseUrl,
          debugTraces: options?.experimental_debugTraces
        })
      );
      Object.keys(import_embedder.SUPPORTED_MODELS).forEach(
        (name) => (0, import_embedder.defineGoogleAIEmbedder)(ai, name, { apiKey: options?.apiKey })
      );
    }
    if (options?.models) {
      for (const modelOrRef of options?.models) {
        const modelName = typeof modelOrRef === "string" ? modelOrRef : (
          // strip out the `googleai/` prefix
          modelOrRef.name.split("/")[1]
        );
        const modelRef = typeof modelOrRef === "string" ? (0, import_gemini.gemini)(modelOrRef) : modelOrRef;
        (0, import_gemini.defineGoogleAIModel)({
          ai,
          name: modelName,
          apiKey: options?.apiKey,
          baseUrl: options?.baseUrl,
          info: {
            ...modelRef.info,
            label: `Google AI - ${modelName}`
          },
          debugTraces: options?.experimental_debugTraces
        });
      }
    }
  });
}
var src_default = googleAI;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  gemini,
  gemini10Pro,
  gemini15Flash,
  gemini15Flash8b,
  gemini15Pro,
  gemini20Flash,
  gemini20FlashExp,
  gemini20FlashLite,
  gemini20ProExp0205,
  gemini25FlashPreview0417,
  gemini25ProExp0325,
  gemini25ProPreview0325,
  googleAI,
  textEmbedding004,
  textEmbeddingGecko001
});
